{
  KettleDatabaseException kettleDatabaseException=new KettleDatabaseException("Unable to check if table [" + EXISTING_TABLE_NAME + "] exists on connection ["+ TEST_NAME_OF_DB_CONNECTION+ "].",SQL_EXCEPTION);
  try {
    when(dbMetaMock.getName()).thenReturn(TEST_NAME_OF_DB_CONNECTION);
    when(resultSetMock.next()).thenReturn(true,false);
    when(resultSetMock.getString("TABLE_NAME")).thenThrow(SQL_EXCEPTION);
    when(dbMetaDataMock.getTables(any(),anyString(),anyString(),aryEq(TABLE_TYPES_TO_GET))).thenReturn(resultSetMock);
    Database db=new Database(mockLogger(),dbMetaMock);
    db.setConnection(mockConnection(dbMetaDataMock));
    db.checkTableExistsByDbMeta(SCHEMA_TO_CHECK,EXISTING_TABLE_NAME);
    fail("There should be thrown KettleDatabaseException but was not.");
  }
 catch (  KettleDatabaseException e) {
    assertTrue(e instanceof KettleDatabaseException);
    assertEquals(kettleDatabaseException.getLocalizedMessage(),e.getLocalizedMessage());
  }
catch (  Exception ex) {
    fail("There should be thrown KettleDatabaseException but was :" + ex.getMessage());
  }
}
